src/building/java/net/remmintan/mods/minefortress/blocks/building/FortressBuildingBlockEntity.kt

```diff
@@ -129,6 +129,12 @@ class FortressBuildingBlockEntity(pos: BlockPos?, state: BlockState?) :
     override fun getFurnace(): FurnaceBlockEntity? =
         furnaceBlockPos?.let { this.getWorld()?.let { w -> w.getBlockEntity(it) as? FurnaceBlockEntity } }
 
+    override fun getFurnacePos(): BlockPos? =
+        this.blockData?.referenceState
+            ?.filter { it.blockState.block?.translationKey == "block.minecraft.furnace" }
```

Comment:
why did you chose such an exotic way of checking if the block is Furnance.
```suggestion
            ?.filter { it.blockState?.isOf(Blocks.FURNACE) }
```

also works. or am I missing something?

Reply:
Nope, you're not missing anything. I'm very new to Kotlin/Java... When I was tracing through with breakpoints and viewing objects in the debugger, I noticed that `translationKey` held the block type I could use, so I just used it. 

I moved over to your better/cleaner implementation in my updated commits.